mlrun/serving/system_steps.py

```diff
@@ -168,6 +168,12 @@ def do(self, event):
                     request, resp = self.reconstruct_request_resp_fields(
                         event, model, monitoring_data[model]
                     )
+                    if hasattr(event, "_original_timestamp"):
+                        when = event._original_timestamp
+                    else:
+                        when = event._metadata.get(model, {}).get(
+                            mm_schemas.StreamProcessingEvent.WHEN
+                        )
```

Comment:
So according to that each batch will have the same timestamp, right?
If it does we should write it somewhere Or better to fully cover this in the stream pod.

Reply:
You're right, `batching` + `timestamp_column` isn't handled properly. Maybe the best way to handle this is by applying the first timestamp to the whole batch, but I think I need to check with @theSaarco.

Reply:
OK @theSaarco, @assaf758  please have a look :)